src/components/MonacoEditor/index.tsx

```diff
@@ -35,9 +27,29 @@ export const MonacoEditor = () => {
   const getHasChanges = useFile(state => state.getHasChanges);
   const theme = useStored(state => (state.lightmode ? "light" : "vs-dark"));
   const fileType = useFile(state => state.format);
+  const editorOptions = React.useMemo(() => {
+    return {
+      formatOnPaste: fileType === FileFormat.JSON5 ? false : true,
+      formatOnType: fileType === FileFormat.JSON5 ? false : true,
+      minimap: {
+        enabled: false,
+      },
+    };
+  }, [fileType]);
 
   React.useEffect(() => {
+    monaco?.languages.typescript.javascriptDefaults.setCompilerOptions({
+      ...monaco.languages.typescript.javascriptDefaults.getCompilerOptions(),
+      checkJs: false,
+      allowUnusedLabels: true,
+    });
+    monaco?.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
+      noSemanticValidation: true,
+      noSyntaxValidation: true,
+    });
+
     monaco?.languages.json.jsonDefaults.setDiagnosticsOptions({
+      ...monaco?.languages.json.jsonDefaults.diagnosticsOptions,
```

Comment:
What is the purpose of this change here?

Reply:
I am adding a comment in the code for this. JSON5 has valid JS syntax, but because the files don't have any exports, it triggers some warnings in Monaco UI if this setting is not there